syntax = "proto3";

message ClientMessages {
	repeated ClientMessage messages = 1;
}

message ClientMessage {
	string messageId = 1;
	MessageData messageData = 2;
}

message MessageData {
	string playerId = 1;
	repeated MessageSource sentBy = 2;
	string messageType = 3;
	string campaignId = 4;
	string messageTitle = 5;
	string messageBody = 6;
	oneof DataAttachment {
		ProgressionMessageDataAttachment progressionCompleteDataAttachment = 7;
	}
	string sentDate = 8;
	string readDate = 9;
	string expirationDate = 10;
}

message MessageSource {
	string type = 1;
	string value = 2;
}

message ProgressionMessageDataAttachment {
	string catalogId = 1;
	string instancedId = 2;
	repeated MessageGrantedRewardData grantedRewardData = 3;
}

message MessageGrantedRewardData {
	string catalogId = 1;
	string amount = 2;
	double sourceobjectiveidx = 3;
}